File: iOS/CoachFit/CoachFit/Models/FoodLogEntry.swift
import Foundation
import SwiftData

@Model
final class FoodLogEntry {
    var date: String
    var barcode: String?
    var name: String
    var brand: String?
    var servingGrams: Double
    var calories: Double
    var protein: Double
    var fat: Double
    var carbs: Double
    var sugar: Double?
    var fiber: Double?
    var sodium: Double?
    var loggedAt: Date

    init(date: String, barcode: String? = nil, name: String, brand: String? = nil,
         servingGrams: Double, calories: Double, protein: Double, fat: Double,
         carbs: Double, sugar: Double? = nil, fiber: Double? = nil,
         sodium: Double? = nil, loggedAt: Date = .now) {
        self.date = date
        self.barcode = barcode
        self.name = name
        self.brand = brand
        self.servingGrams = servingGrams
        self.calories = calories
        self.protein = protein
        self.fat = fat
        self.carbs = carbs
        self.sugar = sugar
        self.fiber = fiber
        self.sodium = sodium
        self.loggedAt = loggedAt
    }
}

File: android/app/src/main/java/com/askadam/coachfit/data/local/FoodLogEntry.kt
package com.askadam.coachfit.data.local

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "food_log")
data class FoodLogEntry(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val date: String,           // YYYY-MM-DD
    val barcode: String?,
    val name: String,
    val brand: String?,
    val servingGrams: Double,
    val calories: Double,
    val protein: Double,
    val fat: Double,
    val carbs: Double,
    val sugar: Double?,
    val fiber: Double?,
    val sodium: Double?,
    val loggedAt: Long           // epoch millis
)
